### ISSUE TITLE
Support skipping of diffImage

### ISSUE THREAD

**Issue by LinusU** [uber/image-diff#23]
I would just like to determine if two pictures looks the same and would not wan't to output an extra image.

I propose that one should be able to skip the `diffImage` parameter and `image-diff` should then output it to a temporary image.

If you want to be correctly guarded against race conditions you shouldn't use a library that generates a random path, but rather a library that creates the file with `O_CREAT | O_EXCL`. A library that does this is my own [fs-temp](https://github.com/LinusU/fs-temp).

```javascript
var temp = require('fs-temp').template('%s.png')

// ...

if (!options.imageDiff) {
  options.imageDiff = temp.writeFileSync('')
}
```

**Comment by twolfson** [comment]
After doing some research, it looks like `imagemagick` supports outputting the `diffImage` to `stdout`:

```bash
# No output image is generated but piped to `/dev/null`
 compare -verbose -metric RMSE -highlight-color RED -compose Src checkerboard.png white.png - 1> /dev/null
```

We can prob coerce `options.diffImage` to `-` when it's falsy to handle this case. However, then we will start to generate a buffer of the diff image which is wasteful (`child_process.exec` doesn't support ignoring this):

https://nodejs.org/api/child_process.html#child_process_child_process_exec_command_options_callback

As a result, we should switch to `child_process.spawn` and use `ignore` for `stdio` on `stdout`:

https://nodejs.org/api/child_process.html#child_process_child_process_spawn_command_args_options

https://nodejs.org/api/child_process.html#child_process_options_stdio

**Comment by twolfson** [comment]
@LinusU I have opened #24 which should fix this issue

**Comment by LinusU** [comment]
:+1: Thank you, looks good

**Comment by twolfson** [comment]
@LinusU This has been patched/released in 1.1.0.

@mlmorg I forgot to add a "Fixes #23" to the PR, can we close this issue? =/
